zomes/elements/code/src/lib.rs

```diff
@@ -136,31 +137,37 @@ fn handle_get_element_index(address: Address) -> ZomeApiResult<ElementIndex> {
     }
 }
 
-fn handle_get_all_games() -> ZomeApiResult<Vec<Element>> {
-    let index_address: Address = handle_create_element_index("Games", "Game").unwrap();
-    let links: Vec<Address> = get_links(&index_address, Some("Index".to_string()), None)?.addresses();
-    let games: Vec<Element> = links.into_iter().map(|address| {
-        handle_get_element(address).unwrap()
-    }).collect();
-    Ok(games)
-}
+fn handle_get_all_elements_of_type(index_type: String) -> ZomeApiResult<Vec<Element>> {
+    let index_name: String = index_type.clone() + " Index";
+    let index = ElementIndex {
+        name: index_name.clone(),
+        type_: index_type
+    };
+
+    let index_entry = Entry::App("ElementIndex".into(), index.into());
+    let address: Address = hdk::commit_entry(&index_entry)?;
 
-fn handle_get_all_formats() -> ZomeApiResult<Vec<Element>> {
-    let index_address: Address = handle_create_element_index("Formats", "Format").unwrap();
-    let links: Vec<Address> = get_links(&index_address, Some("Index".to_string()), None)?.addresses();
-    let formats: Vec<Element> = links.into_iter().map(|address| {
+    let links: Vec<Address> = get_links(&address, Some("Index".to_string()), None)?.addresses();
+    let elements: Vec<Element> = links.into_iter().map(|address| {
         handle_get_element(address).unwrap()
     }).collect();
-    Ok(formats)
+    Ok(elements)
 }
 
-fn handle_get_all_components() -> ZomeApiResult<Vec<Element>> {
-    let index_address: Address = handle_create_element_index("Components", "Component").unwrap();
-    let links: Vec<Address> = get_links(&index_address, Some("Index".to_string()), None)?.addresses();
-    let components: Vec<Element> = links.into_iter().map(|address| {
+fn handle_get_elements_from_index(index_type: String, index_name: String) -> ZomeApiResult<Vec<Element>> {
```

Comment:
What about the term "catalog"?

Reply:
and maybe "definition" or "record" instead of "element". That actually feels a little bit more accurate given that it's a metadata store, it is a record or definition of the element not the element itself.